### DISCUSSION TITLE
Addition of Bluelink/Hyundai Brazil

### DISCUSSION THREAD

**Discussion by lucianoberger** [Hyundai-Kia-Connect/kia_uvo#647] [Ideas]
Hello! I would like to request that the Bluelink Brasil system be added here. How can I help with this problem?

**Comment by cdnninja** [comment]
We will need someone to sniff the traffic and provide it.

**Reply by rafacla** [reply]
Hey guys, I can help, just need basic instructions on how to do this!

**Reply by dburger2023** [reply]
> We will need someone to sniff the traffic and provide it.

I can also help. Just need instructions too

**Reply by vanessa** [reply]
Happy to help with that as well!

**Reply by vanessa** [reply]
I sniffed and saved a couple of requests that the Bluelink app made to `br-ccapi.hyundai.com.br`. What's the best way to provide the requests and responses to the maintainers?

I'm also happy to take a stab at it, would just love some initial pointers on how to best integrate the existing solution with a completely different API endpoint.

**Reply by cdnninja** [reply]
Key will be starting with comparing the requests to each region, starting with EU/ ApiImplType1.py that is the most common one.  If it matches using "stamps" and such may be somewhat easy to add.    If it doesn't match any region would need to start from scratch. 

https://github.com/Hyundai-Kia-Connect/hyundai_kia_connect_api/tree/master/hyundai_kia_connect_api

**Reply by vanessa** [reply]
Thankfully the Brazilian API is fairly similar to India's implementation so I'm working on a POC for the `hyundai_kia_connect_api` package (no ETA yet).

**Comment by vanessa** [comment]
@lucianoberger @dburger2023 opened a draft PR in case someone wants to start reviewing it: https://github.com/Hyundai-Kia-Connect/hyundai_kia_connect_api/pull/814

I still need to address some stuff like locking/unlocking, climate, etc. but the authentication and read-only endpoints are already working.

**Reply by nemiro-TECH** [reply]
Hi Vanessa, did you manage to make it work?
Do you need any help troubleshooting or testing?

**Reply by vanessa** [reply]
Hi, it's working as far as I can tell (based on running the tests). I just need to figure out how to install my branch to Home Assistant with this integration.

**Reply by nemiro-TECH** [reply]
It is returning an error in your pull request.

`
Process completed with exit code 2.
`

**Reply by vanessa** [reply]
Can you give more context? Was this the only error returned?

**Comment by nemiro-TECH** [comment]
Is this topic still on going?
I am from BR and very interested in making this work in my HA.
